js/app.js

```diff
@@ -20,36 +20,64 @@ window.onblur = function() {
 
 var Polyglot = require('node-polyglot'),
     getBTPrice = require('./utils/getBitcoinPrice'),
+    isLocalServerRunning = require('./utils/isLocalServerRunning'),
+    isRemoteServerRunning = require('./utils/isRemoteServerRunning'),
     router = require('./router'),
-    newRouter,
     userModel = require('./models/userMd'),
     userProfileModel = require('./models/userProfileMd'),
     languagesModel = require('./models/languagesMd'),
+    ServerConfigMd = require('./models/serverConfigMd'),
     mouseWheel = require('jquery-mousewheel'),
     mCustomScrollbar = require('./utils/jquery.mCustomScrollbar.js'),
     setTheme = require('./utils/setTheme.js'),
     pageNavView = require('./views/pageNavVw'),
     chatAppView = require('./views/chatAppVw'),
-    newPageNavView,
-    newSocketView,
-    newChatAppView,
     user = new userModel(),
     userProfile = new userProfileModel(),
     languages = new languagesModel(),
     socketView = require('./views/socketVw'),
     cCode = "",
-    serverUrlLocal = "",
-    loadProfileCount = 1,
-    loadProfileCountdownInterval,
-    loadProfileCountdown = 5;
+    $loadingModal = $('.js-loadingModal'),
+    ServerConnectModal = require('./views/serverConnectModal'),
+    OnboardingModal = require('./views/onboardingModal'),
+    serverConfigMd,
+    newRouter,
+    newPageNavView,
+    newSocketView,
+    newChatAppView,
+    serverConnectModal,
+    onboardingModal,
+    startInitSequence,
+    startLocalInitSequence,
+    startRemoteInitSequence,
+    launchOnboarding,
+    launchServerConnect,
+    setServerUrl;
+
+
+// TODO: what is wrong with the localStorage adapter??? shouldn't need
+// to manually provide the data to the model. All that should be needed
+// is an ID and then a subsequent fetch, but that doesn't return the data.
+// Investigate!
+serverConfigMd = new ServerConfigMd( JSON.parse(localStorage['_serverConfig-1']) );
```

Comment:
If localStorage['_serverConfig-1'] comes back undefined, it throws an error and leaves the client stuck at this step. Check to make sure it exists, and set it to "" if it doesn't before this step.